test/e2e/framework/framework.go

```diff
@@ -39,6 +39,19 @@ const (
 	serviceAccountTokenSecretWaitTimeout = 1 * time.Minute
 )
 
+type FrameworkSet []*Framework
```

Comment:
Some of the methods on the framework, like logging or default resources are global and I don't see how forking it per cluster fits there. Maybe we should have just one global framework and use it like `f.Cluster(2).KubeClient()` and `f.KubeClient()` would be a wrapper to redirect it to cluster `0`.

Reply:
I like it, let me try

Reply:
@tnozicka please have a look if this makes sense, especially how the artifacts are structured: https://gcsweb.scylla-operator.scylladb.com/gcs/scylla-operator-prow/pr-logs/pull/scylladb_scylla-operator/1881/pull-scylla-operator-master-e2e-gke-parallel/1788683762628825088/artifacts/

Reply:
roughly yes, I'll go into naming next week. I think it would be beneficial to make an interface for the "Cluster" and make sure the framework implements it fully (by 0 redirection).

artifacts wise, I'd expect cluster index to come before namespace name as it is more significant but a cluster level folder would be even nicer. You also have to run multiple must-gathers

Reply:
> I think it would be beneficial to make an interface for the "Cluster" and make sure the framework implements it fully (by 0 redirection) so the methods don't get out of sync.

good point, I'll add that

> artifacts wise, I'd expect cluster index to come before namespace name as it is more significant

that's a framework name + cluster idx now. I think the framework name is more important, since it corresponds to a test set?

>  but a cluster level folder would be even nicer

So <artifacts-dir>/e2e/<framework-name>/<cluster-name>/...?

Reply:
Actually, thinking about this the second time - when you see a failure in e2e logs you are most likely want to search by the namespace name (statistically unique even across clusters and frameworks) so not nesting things would be easier for the main use case. 
(Instead of 2 jumps for searching the framework name + searching the namespace name in that result.)

`<artifacts-dir>/e2e/{namespaces,cluster-scoped}/<framework-name>-<cluster-name>-<resource-name>`

Reply:
> for the main use case.
> (Instead of 2 jumps for searching the framework name + searching the namespace name in that result.)
> 
> <artifacts-dir>/e2e/{namespaces,cluster-scoped}/<framework-name>-<cluster-name>-<resource-name>

wouldn't that make it quite difficult to find correlated namespaces in multi-datacenter runs? You'd have to run `find` I guess? Although you'd get the other namespaces from the logs, so maybe that's not necessary altogether.

Also I'm not a fan of the prefixes - seems like a lot of redundant information, since all resources in that namespaces are going to come from the same framework/cluster, aren't they? Maybe we could annotate/label the resources with these or just create metadata file(s) there?

Reply:
> wouldn't that make it quite difficult to find correlated namespaces in multi-datacenter runs? 

won't they be sorted out for you by the common prefix?

> Also I'm not a fan of the prefixes - seems like a lot of redundant information

it's technically the same amount of information, you just change the `/` separator to `-`

> since all resources in that namespaces are going to come from the same framework/cluster, aren't they?

are you sure you've read the syntax right? there is a difference between `namespaces` and `<namespace>`

it's similar to what we do today (with cluster index added)

https://gcsweb.scylla-operator.scylladb.com/gcs/scylla-operator-prow/pr-logs/pull/scylladb_scylla-operator/1901/pull-scylla-operator-master-e2e-gke-parallel/1788824474968133632/artifacts/e2e/namespaces/

Reply:
> won't they be sorted out for you by the common prefix?

discard my comment, over the weekend I managed to forget the namespaces were generated with framework/cluster names as prefix

> are you sure you've read the syntax right? there is a difference between namespaces and <namespace>

I first understood your syntax as such that every leaf resource would have this prefix. Nevermind - I updated it now, I suppose we're on the same page.

Reply:
we are, thx